rdflib/namespace/__init__.py

```diff
@@ -590,6 +590,31 @@ def compute_qname_strict(
 
             return self.__cache_strict[uri]
 
+    def expand_curie(self, curie: str) -> Union[URIRef, None]:
+        """
+        Expand a qname (aka CURIE) of the form <prefix:element>, e.g. "rdf:type"
+        into its full expression:
+
+        >>> import rdflib
+        >>> g = rdflib.Graph()
+        >>> g.namespace_manager.expand_curie("rdf:type")
+        'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+
+        Returns None if a namespace is not bound to the prefix.
+
+        """
+        if not type(curie) is str:
+            raise TypeError("Argument must be a string.")
+        elif ":" not in curie or curie.split(":")[0] == "" or curie.split(":")[1] == "":
+            raise ValueError(
+                "Malformed curie argument, format should be e.g. “foaf:name”."
+            )
```

Comment:
The follow are valid curies which will not be be accepted, and resolve as follow given `@prefix ex: <https://example.com/>.`:

- `ex:` -> `https://example.com/`
- `ex:a:b:c` -> `https://example.com/a:b:c`

Checked with riot:

```console
$ echo "@prefix ex: <https://example.com/>. ex: ex:a ex:a:b:c." | riot --syntax TURTLE --out NTRIPLES
<https://example.com/> <https://example.com/a> <https://example.com/a:b:c> .
```

I will make a PR to your branch shortly to add this, and to also reduce the amount of times split is called.

Reply:
Will also fix test in it, I simplified them a bit.

Reply:
Pull request here: https://github.com/gjhiggins/rdflib/pull/6/files

I would be very happy to apply this to your branch directly, just let me know.

Reply:
Applied now.